src/components/UI/Header/Header.js

```diff
@@ -6,18 +8,31 @@ import {
   faCartShopping,
   faChair,
 } from '@fortawesome/free-solid-svg-icons';
+import Search from '../../Search';
 
 const Header = () => {
+  const [isSearchModalOpen, setIsSearchModalOpen] = useState(false);
+
   const openSearchModalHandler = () => {
-    console.log('open search modal');
+    setIsSearchModalOpen(true);
+  };
+
+  const closeSearchModalHandler = () => {
+    setIsSearchModalOpen(false);
   };
 
   const openCartModalHandler = () => {
     console.log('open cart modal');
   };
 
+  const searchModal = ReactDOM.createPortal(
+    <Search closeModal={closeSearchModalHandler} />,
+    document.getElementById('modal')
+  );
+
```

Comment:
instead of using portal multiple times you can create separate component. 
Or you can consider using `react-modal` package from npm

Reply:
I created one component Modal wchich render element from props as portal